### ISSUE TITLE
Verify consistency of the Context

### ISSUE THREAD

**Issue by ppamorim** [fabiendevos/nanotasks#5]

**Comment by fabiendevos** [comment]
What do you mean by consistency?

**Comment by ppamorim** [comment]
If Activity isn't finishing or fragment is added. These are things that can be implemented out-of-box, but it is interesting to implement them...

**Comment by fabiendevos** [comment]
Yes I want to provide better checking / handling of the life cycle. I'll try to do it in a way that's very simple for the user of the lib, "do the right thing" out of the box, and is difficult to get wrong.
